include/IECoreScene/SceneInterface.h

```diff
@@ -251,6 +269,8 @@ class IECORESCENE_API SceneInterface : public IECore::RunTimeTyped
 		/// Returns a const interface for querying the scene at the given path (full path).
 		virtual ConstSceneInterfacePtr scene( const Path &path, MissingBehaviour missingBehaviour = ThrowIfMissing ) const = 0;
 
+		/// Returns the root SceneInteface for this heirahcy.
+		SceneInterfacePtr getRoot() const { return m_root; }
```

Comment:
> As I've explained before, in Cortex and Gaffer, set/get are always paired and imply mutability of a member.

Let's hope you never have to mention this again m _ _ m.

> That said, I don't get the need for this method - how is it different to calling this->scene( Path() )?

I added this for the reading tags in the LinkedScene case. What happens when I want to read a tag on a linked scene which was saved with v10?

Reply:
I'm afraid I don't understand the details here. But lower down you say "Looks like I can do this - thanks for pointing it out.", in reference to the same question, so I'm a bit confused.

Reply:
>  I'm afraid I don't understand the details here. But lower down you say "Looks like I can do this - thanks for pointing it out.", in reference to the same question, so I'm a bit confused.

I took another look at this->scene( Path() ) and I don't need to keep a root SceneInterface. We do require every class derived from SceneInterface to do something like keeping reference to the root or something similar.

Reply:
> I took another look at this->scene( Path() ) and I don't need to keep a root SceneInterface.

Ah, good! Does that also take care of your const-correctness issues, since there's a const and a non-const `scene()`, or is that separate?

> We do require every class derived from SceneInterface to do something like keeping reference to the root or something similar.

Yep, they need something similar in order to implement `scene()`. But they've all avoided using a SceneInterfacePtr to the root, even though that might have been the obvious choice, so I do wonder if that was for good reason.

Reply:
> Ah, good! Does that also take care of your const-correctness issues, since there's a const and a non-const scene(), or is that separate?

I think this will be OK, but as you've pointed out there are more important issues I should address first.

> Yep, they need something similar in order to implement scene(). But they've all avoided using a SceneInterfacePtr to the root, even though that might have been the obvious choice, so I do wonder if that was for good reason.

Each concrete implementation of SceneInterface needs to either store some shared data or pointer to parent. For ::scene to be implemented we in SceneCache we need to walk parent pointers back to the root. https://github.com/donboie/cortex/blob/e160d557e46b64e2cc496f026415a5fde297c272/src/IECoreScene/SceneCache.cpp#L585

Reply:
Yep, but for instance, Alembic Scene stores things the other way, and each location owns a map of its children :

https://github.com/ImageEngine/cortex/blob/master/contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp#L555

Now it's not storing a map of SceneInterfacePtr so I think it would be OK, but if it did want to (and that doesn't seem unreasonable), then the root pointer would preclude that because it would cause a circular reference...